tests: gen_isr_table: do not plug ISR4 for TI cc13x2/cc26x2

vanti opened the pull request:
This test blindly tries to plug IRQ 34 (ISR4), which is being plugged
in the dpl layer. This causes a build failure, so we are changing the
test to not plug this interrupt.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

ioannisg commented:
@vanti @galak I wonder if we should treat this as a Hotfix (added the label anyways)

stephanosio commented:
> @vanti @galak I wonder if we should treat this as a Hotfix (added the label anyways)

This did break CI for the PR #21326, so I suppose that would only make sense.

ioannisg reviewed (commented):
I am OK with the change; this test is really messy, anyways, and needs re-work, so this patch does not make it worse.

One question @vanti : why didn't you simply exclude the specific platform(s) from the test? We do it for other SoCs.

Finally, if you still want to run the test for this TI platform, may I ask you whether the test is passing on target? I have no such board to test it.

vanti commented:
@ioannisg I worked under the assumption that the more we test, the better. The test does run successfully after the change on the cc1352r1_launchxl, so I figured we might as well make use of it.

nashif commented on tests/kernel/gen_isr_table/src/main.c:
```diff
@@ -28,7 +28,9 @@ extern u32_t _irq_vector_table[];
 #define TRIG_CHECK_SIZE		6
 #else
 #define ISR3_OFFSET	2
+#if !defined(CONFIG_SOC_SERIES_CC13X2_CC26X2)
```
we should avoid having SoC/Board specific Kconfigs in tests, this will get very ugly when more SoCs/Boards are excluded like this. Please exclude it in testcase.yaml or actually we should whitelist this test just for boards that do not have conflicts in ISRs.

ioannisg replied:
I've also suggested that as an option.

However, I also see the point of testing as-much-as-we-can on each board. This test needs serious re-work anyways, eventually we will re-write this.

In brief, I am OK with both approaches.

carlocaione replied:
Since this is literally blocking #21326 (and #21326 in cascade) can I propose to merge this as is for now?

vanti replied:
@nashif @ioannisg I excluded the test from testcase.yaml as the simplest way to move this along, given the push back on having platform specific code. It is not entirely clear to me which boards need to be whitelisted, so I won't take that approach.

andrewboie replied:
@nashif FWIW I am ok with this change, this test honestly needs to be scrapped and rewritten

vanti committed:
tests: kernel: gen_isr_table: exclude from testcase.yaml

Excluding this test for cc13x2/cc26x2 platforms where ISR4 is not
available for it to use.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>
